aztk/client.py

```diff
@@ -159,6 +160,73 @@ def __get_remote_login_settings(self, pool_id: str, node_id: str):
         result = self.batch_client.compute_node.get_remote_login_settings(pool_id, node_id)
         return models.RemoteLogin(ip_address=result.remote_login_ip_address, port=str(result.remote_login_port))
 
+    def __submit_job(self, job_configuration, start_task, job_manager_task, autoscale_formula, software_metadata_key: str, vm_image_model):
+        """
+            Job Submission
+            :param job_configuration -> aztk_sdk.spark.models.JobConfiguration
+            :param start_task -> batch_models.StartTask
+            :param job_manager_task -> batch_models.TaskAddParameter
+            :param autoscale forumula -> str
+            :param software_metadata_key: str
+            :param vm_image_model -> aztk_sdk.models.VmImage
+            :returns None
+        """
+        # get a verified node agent sku
+        sku_to_use, image_ref_to_use = \
+            helpers.select_latest_verified_vm_image_with_node_agent_sku(
+                vm_image_model.publisher, vm_image_model.offer, vm_image_model.sku, self.batch_client)
+
+        # set up a schedule for a recurring job
+        auto_pool_specification = batch_models.AutoPoolSpecification(
+            pool_lifetime_option=batch_models.PoolLifetimeOption.job_schedule,
+            auto_pool_id_prefix=job_configuration.id,
+            keep_alive=False,
+            pool=batch_models.PoolSpecification(
+                display_name=job_configuration.id,
+                virtual_machine_configuration=batch_models.VirtualMachineConfiguration(
+                    image_reference=image_ref_to_use,
+                    node_agent_sku_id=sku_to_use),
+                vm_size=job_configuration.vm_size,
+                enable_auto_scale=True,
+                auto_scale_formula=autoscale_formula,
+                start_task=start_task,
+                enable_inter_node_communication=True,
+                max_tasks_per_node=1,
+                metadata=[
+                    batch_models.MetadataItem(
+                        name=constants.AZTK_SOFTWARE_METADATA_KEY, value=software_metadata_key),
+                ]
+            )
+        )
+
+        # define job specification
+        job_spec = batch_models.JobSpecification(
+            pool_info=batch_models.PoolInformation(auto_pool_specification=auto_pool_specification),
+            display_name=job_configuration.id,
+            on_all_tasks_complete=batch_models.OnAllTasksComplete.terminate_job,
+            job_manager_task=job_manager_task
+        )
+
+        # define schedule
```

Comment:
Probably fine for now since we're not supporting anything but 'run now' but I would expect this is something that would be passed in as a parameter in the future.

Reply:
Originally, the JobConfiguration model had a fields for each of these values. Since we are going with just 'run now' for the time being, I removed those. 

When we add scheduling functionality, we should probably add those fields (or an abstraction layer for those fields) back to the JobConfiguration model.